libs/category/testing/src/inmemory-backend/category.service.ts

```diff
@@ -63,30 +64,21 @@ export class DaffInMemoryBackendCategoryService implements InMemoryDbService {
     });
 	}
 	
-	private getTotalPages(allIds: number[], pageSize: number) {
-		return Math.floor(allIds.length / pageSize) + 1;
+	private getTotalPages(allIds: string[], pageSize: number) {
+		return allIds.length % pageSize ? Math.floor(allIds.length / pageSize) + 1: allIds.length/pageSize;
 	}
 
-	private trimProductIdsToSinglePage(allIds: number[], currentPage: number, pageSize: number) {
+	private trimProductIdsToSinglePage(allIds: string[], currentPage: number, pageSize: number) {
 		const tempIds = [...allIds];
 		tempIds.splice(0, (currentPage-1) * pageSize);
 		tempIds.splice(pageSize, tempIds.length-pageSize);
 
 		return tempIds;
 	}
 
-	private generateProductIdArray(products: DaffProduct[]) {
-		const arraySize = Math.floor(Math.random() * Math.floor(products.length/2) + Math.floor(products.length/2));
-		const productIdArray = [];
-		for(let i=0; i<arraySize; i++) {
-			const productIndex = Math.floor(Math.random() * products.length);
-			productIdArray.push(
-				products[productIndex].id
-			);
-			products.splice(productIndex, 1);
-		}
-
-		return productIdArray;
+	private generateProductIdSubset(products: DaffProduct[]): string[] {
+		const subsetSize = Math.floor(Math.random() * Math.floor(products.length/2) + Math.floor(products.length/2));
```

Comment:
I don't think we should be guaranteeing the size here. I'd like to be able to get small, or 0 products in a response.

Reply:
done